packages/datadog-plugin-openai/src/index.js

```diff
@@ -157,6 +157,10 @@ class OpenApiPlugin extends TracingPlugin {
   }
 
   finish ({ headers, body, method, path }) {
+    try {
+      headers = Object.fromEntries(headers)
+    } catch { /* headers are already an object */ }
```

Comment:
I'd rather we _check_ it's an object rather than try/catching our way out of this.